rent_golem/backend/rent_golem/clusters/tests.py

```diff
@@ -1,3 +1,225 @@
+        self.assertTrue(Cluster.objects.filter(uuid=cluster.uuid).exists())
+        self.assertEquals(Cluster.objects.filter(uuid=cluster.uuid).first().size, 6)
+
+
+    def test_update_by_owner_2_variables(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        self.client.force_authenticate(user=self.owner)
+
+        update_response = self.client.patch(
+            reverse('cluster-detail', args=(cluster.uuid,)),
+            data={'size': cluster.size + 1, 'package_type': Cluster.Package.AUTOMATIC}
+        )
+
+        self.assertEquals(update_response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertTrue(Cluster.objects.filter(uuid=cluster.uuid).exists())
+        self.assertEquals(Cluster.objects.filter(uuid=cluster.uuid).first().size, 5)
+
+    def test_update_by_user_not_owner(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        jwt_insignificant_user = self.client.post(
+            reverse("token_obtain_pair"), data={"email": self.insignificant_user.email, "password": "test1234"}
+        ).data["access"]
+
+        update_response = self.client.patch(
+            reverse('cluster-detail', args=(cluster.uuid, )),
+            data={'size': cluster.size+1},
+            headers={"Authorization": f"Bearer {jwt_insignificant_user}"}
+        )
+
+        self.assertEquals(update_response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertTrue(Cluster.objects.filter(uuid=cluster.uuid).exists())
+        self.assertEquals(Cluster.objects.filter(uuid=cluster.uuid).first().size, 5)
+
+    def test_update_by_anonymous_user(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        update_response = self.client.patch(
+            reverse('cluster-detail', args=(cluster.uuid, )),
+            data={'size': cluster.size+1}
+        )
+
+        self.assertEquals(update_response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertTrue(Cluster.objects.filter(uuid=cluster.uuid).exists())
+        self.assertEquals(Cluster.objects.filter(uuid=cluster.uuid).first().size, 5)
+
+    def test_put_turned_off(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        self.client.force_authenticate(user=self.owner)
+
+        put_response = self.client.put(
+            reverse('cluster-detail', args=(cluster.uuid, )),
+            data={'size': cluster.size+1}
+        )
+
+        self.assertEquals(put_response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    # Delete
+    def test_delete_by_owner(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        self.client.force_authenticate(user=self.owner)
+
+        delete_response = self.client.delete(reverse('cluster-detail', args=(cluster.uuid,)))
+
+        self.assertEquals(delete_response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertFalse(Cluster.objects.filter(uuid=cluster.uuid).exists())
```

Comment:
get() instead filter().first()

Reply:
We should here check if object is still in DB but has is_deleted flag set to True.